zoo/policies/__init__.py

```diff
@@ -100,3 +108,126 @@ def human_keyboard_entrypoint(*arg, **kwargs):
 
 
 register(locator="human-in-the-loop-v0", entry_point=human_keyboard_entrypoint)
+
+
+def load_config(path):
+    import yaml
+
+    config = None
+    if path.exists():
+        with open(path, "r") as file:
+            config = yaml.safe_load(file)
+    return config
+
+
+root_path = str(Path(__file__).absolute().parents[2])
+
+
+def competition_entry(**kwargs):
+    policy_path = kwargs.get("policy_path", None)
+    comp_env_path = str(
+        os.path.join(policy_path, "competition_env")
+    )  # folder contains all competition environment
+    sub_env_path = os.path.join(
+        comp_env_path, f"{Path(policy_path).name}"
+    )  # folder contains single competition environment
+    req_file = os.path.join(
+        policy_path, "requirements.txt"
+    )  # path of the requiremnet file
+
+    if Path(sub_env_path).exists():
+        shutil.rmtree(sub_env_path)
+    Path.mkdir(Path(sub_env_path), parents=True, exist_ok=True)
+
+    try:
+        subprocess.check_call(
+            [
+                sys.executable,
+                "-m",
+                "pip",
+                "install",
+                "-t",
+                sub_env_path,
+                "-r",
+                req_file,
+            ]
+        )
+        sys.path.append(sub_env_path)
+    except:
+        logging.error(
+            f"Failed to install requirement for Competition Agent in folder {Path(policy_path).name}"
+        )
+        raise
+
+    # insert submission path
+    while policy_path in sys.path:
+        sys.path.remove(policy_path)
+
+    sys.path.insert(0, policy_path)
```

Comment:
Why is this necessary? Is there some sort of leak?
```suggestion
    if policy_path not in sys.path:
        sys.path.insert(0, policy_path)
```

Reply:
Just want to ensure that policy_path is been searched first in the sys.path to avoid potential issues. If the policy is already in sys.path and there are some other paths contain `policy.py` in front of it, it may cause some import errors.

Reply:
Oh, interesting, the imports resolve in the order they occur in `sys.path`? I think I understand now.

Reply:
Will do.